From 7b3fbac52865f73038f3d65f05cde7206f644274 Mon Sep 17 00:00:00 2001
From: Alex Johnston <acjohnston@google.com>
Date: Mon, 29 Nov 2021 16:06:38 +0000
Subject: [PATCH] Add non system overlay flag to InstallCaCertificateWarning

Bug: 196969991
Test: Manual testing with Settings
Merged-In: Ia9dc251c853526b2ce66c9f8ff595d496b7f1bc4
Change-Id: Ia9dc251c853526b2ce66c9f8ff595d496b7f1bc4
(cherry picked from commit 2f193c81576e2e86cf64470aeb312cab1e76d40d)
(cherry picked from commit 02e8f56e67b91b0909c713bcbb949800f244b30a)
Merged-In:Ia9dc251c853526b2ce66c9f8ff595d496b7f1bc4
---
 .../settings/security/InstallCaCertificateWarning.java      | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/com/android/settings/security/InstallCaCertificateWarning.java b/src/com/android/settings/security/InstallCaCertificateWarning.java
index 38548756a3..139bc1c270 100644
--- a/src/com/android/settings/security/InstallCaCertificateWarning.java
+++ b/src/com/android/settings/security/InstallCaCertificateWarning.java
@@ -16,6 +16,8 @@
 
 package com.android.settings.security;
 
+import static android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS;
+
 import android.annotation.Nullable;
 import android.app.Activity;
 import android.content.Intent;
@@ -45,6 +47,8 @@ public class InstallCaCertificateWarning extends Activity {
         setTheme(SetupWizardUtils.getTheme(this, getIntent()));
         ThemeHelper.trySetDynamicColor(this);
         setContentView(R.layout.ca_certificate_warning_dialog);
+        getWindow().addSystemFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         final GlifLayout layout = findViewById(R.id.setup_wizard_layout);
         layout.setHeaderText(R.string.ca_certificate_warning_title);
 
@@ -57,6 +61,7 @@ public class InstallCaCertificateWarning extends Activity {
                         .setTheme(R.style.SudGlifButton_Secondary)
                         .build()
         );
+        mixin.getSecondaryButtonView().setFilterTouchesWhenObscured(true);
 
         mixin.setPrimaryButton(
                 new FooterButton.Builder(this)
@@ -66,6 +71,7 @@ public class InstallCaCertificateWarning extends Activity {
                         .setTheme(R.style.SudGlifButton_Primary)
                         .build()
         );
+        mixin.getPrimaryButtonView().setFilterTouchesWhenObscured(true);
     }
 
     private View.OnClickListener installCaCertificate() {
-- 
2.17.1

